src/main.rs

```diff
@@ -29,42 +69,12 @@ async fn main() {
         Err(_err) => return,
     };
 
-    let provision_with_password = imds::get_provision_with_password(&imds_body);
-    let disable_authentication = match provision_with_password {
-        Ok(disable_authentication) => disable_authentication,
+    let username = match get_username(imds_body.clone()) {
+        Ok(res) => res,
         Err(_err) => return,
     };
 
-    let username;
-    let mut password = "".to_owned();
-
-    if !disable_authentication {
-        media::make_temp_directory().unwrap();
-
-        media::mount_media();
-
-        let ovf_body = media::read_ovf_env_to_string().unwrap();
-        let environment = media::parse_ovf_env(ovf_body.as_str()).unwrap();
-
-        username = environment
-            .provisioning_section
-            .linux_prov_conf_set
-            .username;
-        password = environment
-            .provisioning_section
-            .linux_prov_conf_set
-            .password;
-
-        let _ = media::allow_password_authentication();
-
-        media::remove_media();
-    } else {
-        let username_result = imds::get_username(imds_body.clone());
-        username = match username_result {
-            Ok(username) => username,
-            Err(_err) => return,
-        };
-    }
+    let password = "".to_owned();
```

Comment:
Since password is only provided to the `Distribution::create_user` function it doesn't need to be a `String`:
```suggestion
    let password = "";
```

Of course, if we're not ever going to have passwords the API should probably not even take a `password` argument.

Reply:
Done. Simply removed the `password` variable, and passed `""`.